@here/harp-vectortile-datasource/lib/VectorTileDataEmitter.ts

```diff
@@ -226,6 +226,17 @@ enum LineType {
 type TexCoordsFunction = (tilePos: THREE.Vector2, tileExtents: number) => THREE.Vector2;
 const tmpColor = new THREE.Color();
 
+/**
+ * Options for VectorTileDataEmitter, @see {@link DecoderOptions} and {@link OmvDecoderOptions}.
+ * @internal
+ */
+export interface VectorTileDataEmitterOptions {
```

Comment:
Ideally this should be in a separate commit / PR, because it seems just a refactoring and not related to the actual ticket?

Reply:
True. But since it hardly adds any complexity to the change, may we keep it here? Otherwise I have to rewrite all the usages back to the original signature and then introduce the interface again in a separate PR.

I see you try to reduce the size of this PR, but most of it comes from the unit tests (+600 lines) and then moving OMV specific options out of VectorTileDecoder and VectorTileDataProcessor.
The changes related to custom adapters are in the first two commits. The rest of the commits are unit tests and refactoring.

Reply:
Ok, fair enough.